src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -3548,6 +3595,14 @@ select clausesByQuery.First())
                 {
                     if (!map.Reverse.ContainsKey(newLambda))
                     {
+                        if ((IsLocalFunction(newLambda) && !capabilities.HasCapability(ManagedEditAndContinueCapability.AddDefinitionToExistingType)) ||
+                           (!IsLocalFunction(newLambda) && !capabilities.HasCapability(ManagedEditAndContinueCapability.NewTypeDefinition)))
```

Comment:
This check is not quite right. If the lambda added is the first one in the containing type of the method, then this is fine, but if there is already a lambda then this check would just be `AddDefinitionToExistingType`. Unfortunately I haven't worked out how to nicely determine if a class has a lambda already, so this is a TODO :)

Reply:
Couldn't find a way to do this, so logged https://github.com/dotnet/roslyn/issues/52759 to improve things in a follow up.